src/store/index-level.ts

```diff
@@ -0,0 +1,205 @@
+import type { LevelDatabase } from './create-level.js';
+import type { AbstractBatchDelOperation, AbstractBatchOperation, AbstractIteratorOptions } from 'abstract-level';
+
+import { abortOr } from '../utils/abort.js';
+import { createLevelDatabase } from './create-level.js';
+import flat from 'flat';
+
+const KEY_DELIMITER = '<';
+const KEY_BOUNDARY = '>';
+
+const VALUE_DELIMITER = '=';
+
+const KEY_PREFIX_DELIMITER = '@';
+
+export type Entry = {
+  _id: string,
+  [property: string]: unknown
+};
+
+export type ExactFilter = string | number | boolean;
+
+export type RangeFilter = {
+  gte?: string,
+  lte?: string
```

Comment:
why dont we want to allow for the "or equal" part?  it seems a bit unnecessary (especially at such a generic low-level like this) to restrict it to `lt` only

Reply:
Sorry for omitting the full context: it's because we really should have `to` in `RangeCriterion` currently only used for dates be exclusive of the value specified. This way a dev can iterate through time ranges without needing to add silly dedup logic. This would be inline with convention in JavaScript array operations (and other languages).

Reply:
IMO `RangeCriterion` being end range exclusive is kinda a separate issue to whether this very low level structure can support functionality like this.  i would much prefer having the ability to do this at a low level and then a higher level thing choosing not to use it than not even having the ability at any level.

also, why not provide the ability for developers to do `lt` or `lte`?  it seems a bit unfortunate that we would expect the developer to do this work when we already have the ability to do so

Reply:
I am kinda lost. But this one was an optional request 'coz I thought it was low hanging fruit, but feel free to resolve this as out-of-scope for this PR.